apps/spruce/src/pages/waterfall/WaterfallGrid.tsx

```diff
@@ -141,13 +158,40 @@ export const WaterfallGrid: React.FC<WaterfallGridProps> = ({
   );
 
   const { activeVersionIds, buildVariants, versions } = useFilters({
+    activeVersionIds: data.waterfall.pagination.activeVersionIds,
     buildVariants: groupedBuildVariants,
     flattenedVersions: data.waterfall.flattenedVersions.map(
       ({ waterfallBuilds, ...restOfVersion }) => restOfVersion,
     ),
     pins,
   });
 
+  const [isPending, startTransition] = useTransition();
+  const [allQueryParams] = useQueryParams();
+
+  useEffect(() => {
+    const serverQueryParamsEmpty =
+      !Object.keys(allQueryParams).includes(
+        WaterfallFilterOptions.Requesters,
+      ) &&
+      !Object.keys(allQueryParams).includes(
+        WaterfallFilterOptions.BuildVariant,
+      );
```

Comment:
nit. this is just a matter of preference, but I think reversing the condition might be slightly more readable
```
 const hasServerParams =
      Object.keys(allQueryParams).includes(WaterfallFilterOptions.Requesters) ||
      Object.keys(allQueryParams).includes(WaterfallFilterOptions.BuildVariant);
```